tests/gateway/test_runner.py

```diff
@@ -33,32 +34,39 @@ def __init__(self, config_path: str, *args, **kwargs):
             ],
             *args,
             **kwargs,
-            preexec_fn=os.setsid,
         )
         self.wait_until_ready()
 
     def wait_until_ready(self) -> None:
         s = time.time()
         while time.time() - s < 10:
             try:
-                if self.request("health").ok:
+                if self.get("health").ok:
                     return
             except requests.exceptions.ConnectionError:
-                time.sleep(0.1)
+                time.sleep(0.5)
 
         raise Exception("Gateway failed to start")
 
-    def request(self, path: str) -> requests.Response:
-        return requests.get(f"http://{self.host}:{self.port}/{path}")
+    def request(self, method: str, path: str, *args: Any, **kwargs: Any) -> requests.Response:
+        return requests.request(method, f"http://{self.host}:{self.port}/{path}", *args, **kwargs)
 
-    def post(self, path: str, query: Dict[str, Any]) -> requests.Response:
-        return requests.post(f"http://{self.host}:{self.port}/{path}", json=query)
+    def get(self, path: str, *args: Any, **kwargs: Any) -> requests.Response:
+        return self.request("GET", path, *args, **kwargs)
 
-    def __exit__(self, *args, **kwargs):
-        kill_child_processes(self.pid)
-        os.kill(self.pid, signal.SIGTERM)  # kill the master process first
+    def assert_health(self):
+        assert self.get("health").ok
```

Comment:
When a function asserts, I prefix it with `assert_` for clarity. `check` sounds ambiguous and could mean anything.